[FLINK-28456] Document streaming query page (#207)
apache/flink-table-store: docs/content/docs/development/query-table.md

# Query Table

You can directly SELECT the table in batch runtime mode of Flink SQL.

```sql
-- Batch mode, read latest snapshot
SET 'execution.runtime-mode' = 'batch';
SELECT * FROM MyTable;
```

## Query Engines

Table Store not only supports Flink SQL queries natively but also provides
queries from other popular engines. See [Engines]({{< ref "docs/engines/overview" >}})

## Query Optimization

It is highly recommended to specify partition and primary key filters
along with the query, which will speed up the data skipping of the query.

The filter functions that can accelerate data skipping are:
- `=`
- `<`
- `<=`
- `>`
- `>=`
- `IN (...)`

> - `LIKE '%abc'`

Reviewer: IIUC, we only support start pattern?

Author: Yes, this is the start pattern

- `IS NULL`

> Table Store will sort the data by primary key, which is good for point queries

Reviewer: ```suggestion
Table Store will sort the data by primary key, which speeds up the point
and range queries. When using a composite primary key, the query filters should form a [leftmost prefix](https://dev.mysql.com/doc/refman/5.7/en/multiple-column-indexes.html)
of the primary key for good acceleration.
```

Author: `speeds up the` looks good to me.
But for `the query filters should form`, I think it is not a `should`, it is just an optimization.

and range queries to speed up. When using a composite primary key, it is best
for the query filters to form a [leftmost prefix](https://dev.mysql.com/doc/refman/5.7/en/multiple-column-indexes.html)
of the primary key for good acceleration.

Suppose that a table has the following specification:

```sql
CREATE TABLE orders (
    catalog_id BIGINT,
    order_id BIGINT,
    .....,

>     PRIMARY KEY (catalog_id, order_id) NOT ENFORCED

Reviewer: ```suggestion
    PRIMARY KEY (catalog_id, order_id) NOT ENFORCED -- composite primary key
```

)
```

> The primary key is catalog_id + order_id. Good acceleration is obtained

Reviewer: ```suggestion
The query obtains a good acceleration by specifying a range filter for the leftmost prefix of the primary key.
```

by specifying a range filter for the leftmost prefix of the primary key.

```sql
SELECT * FROM orders WHERE catalog_id=1025;

> SELECT * FROM orders WHERE catalog_id=1025 AND order_id=29495;

Reviewer: format the sql
```suggestion
SELECT * FROM orders WHERE catalog_id = 1025 AND order_id = 29495;
```

Author: When I take a look to https://dev.mysql.com/doc/refman/5.7/en/multiple-column-indexes.html
We don't have to add ` ` in here.

> SELECT * FROM orders
>   WHERE catalog_id=1025
>   AND order_id>2035 AND order_id<6000;

Reviewer: ```suggestion
SELECT * FROM orders
  WHERE catalog_id = 1025
  AND order_id > 2035 AND order_id < 6000;
```

```

However, there will not be a good acceleration in the following query.

```sql

> SELECT * FROM orders WHERE order_id=29495;

Reviewer: ```suggestion
SELECT * FROM orders WHERE order_id = 29495;
```
